lnbits/wallets/nwc.py

```diff
@@ -0,0 +1,881 @@
+            # to call lookup_invoice too (if supported)
+            info = await self._get_info()
+            if "lookup_invoice" in info["supported_methods"]:
+                try:
+                    payment_data = await self._call(
+                        "lookup_invoice", {"invoice": bolt11}
+                    )
+                    settled = payment_data.get("settled_at", None) and payment_data.get(
+                        "preimage", None
+                    )
+                    if not settled:
+                        return PaymentResponse(None, payment_hash, None, None, None)
+                    else:
+                        fee_msat = payment_data.get("fees_paid", None)
+                        return PaymentResponse(
+                            True, payment_hash, fee_msat, preimage, None
+                        )
+                except Exception:
+                    # Workaround: some nwc service providers might not store the invoice
+                    # right away, so this call may raise an exception.
+                    # We will assume the payment is pending anyway
+                    return PaymentResponse(None, payment_hash, None, None, None)
+            else:  # if not supported, we assume it succeeded
+                return PaymentResponse(True, payment_hash, None, preimage, None)
+
+        except Exception as e:
+            return PaymentResponse(ok=False, error_message=str(e))
+
+    async def get_invoice_status(self, checking_id: str) -> PaymentStatus:
+        return await self.get_payment_status(checking_id)
+
+    async def get_payment_status(self, checking_id: str) -> PaymentStatus:
+        try:
+            info = await self._get_info()
+            if "lookup_invoice" in info["supported_methods"]:
+                payment_data = await self._call(
+                    "lookup_invoice", {"payment_hash": checking_id}
+                )
+                settled = payment_data.get("settled_at", None) and payment_data.get(
+                    "preimage", None
+                )
+                fee_msat = payment_data.get("fees_paid", None)
+                preimage = payment_data.get("preimage", None)
+                created_at = int(payment_data.get("created_at", time.time()))
+                expires_at = int(payment_data.get("expires_at", created_at + 3600))
+                expired = expires_at and time.time() > expires_at
+                if expired and not settled:
+                    return PaymentStatus(False, fee_msat=fee_msat, preimage=preimage)
+                else:
+                    return PaymentStatus(
+                        True if settled else None, fee_msat=fee_msat, preimage=preimage
+                    )
+            else:
+                return PaymentStatus(None, fee_msat=None, preimage=None)
+        except Exception as e:
+            logger.error("Error getting payment status: " + str(e))
+            return PaymentFailedStatus()
```

Comment:
This method should not return 'failed' unless the funding source explicitly returns a failed status.
It can get to the catch block due to a network error or an unrelated bug.